core-rust/state-manager/src/mempool/priority_mempool.rs

```diff
@@ -254,9 +254,7 @@ impl PriorityMempool {
         // Check the new transaction is better than all to be removed transactions.
         if !to_be_removed.is_empty() {
             let best_to_be_removed = to_be_removed.last().unwrap();
-            if new_order_data
-                < MempoolDataProposalPriorityOrdering((*to_be_removed.last().unwrap()).clone())
-            {
+            if new_order_data < MempoolDataProposalPriorityOrdering(best_to_be_removed.clone()) {
```

Comment:
_(note from author)_

This is just a drive-by I spotted while reading lots of code related to our metrics.